src/core/server/saved_objects/export/get_sorted_objects_for_export.ts

```diff
@@ -180,6 +180,7 @@ export async function exportSavedObjectsToStream({
     exportedObjects = sortObjects(rootObjects);
   }
 
+  exportedObjects = exportedObjects.filter((obj) => obj.type !== 'data-source');
```

Comment:
Instead of filtering the final results, shall we filter the `types` passed in before sending it to `savedObjectsClient.find`? Or maybe even throw error if it tries to export data-source object, like in `fetchObjectsToExport` we do validations for the export parameters.

Reply:
+1, I assume we are using the saved object find API and we can specify the object types there

Reply:
Sure, + 1 on using saved object find API and specifying the types; as for 

> Or maybe even throw error if export data-source is not allowed
+ 0.5, error can be some interruption on UX, not quite sure about the user experience, but we can have warning first, more discussion is [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6115)